src/racer/typeinf.rs

```diff
@@ -334,6 +334,15 @@ pub fn get_return_type_of_function(fnmatch: &Match, contextm: &Match, session: &
         ast::parse_fn_output(decl, Scope::from_match(fnmatch))
     });
 
+    // Convert output arg of type Self to the correct type
+    if let Some(core::Ty::PathSearch(ref mut path, _)) = out {
+        if let Some(ref mut path_seg) = path.clone().segments.get(0) {
```

Comment:
Ah yeah. Without the `ref` in `if let Some(ref thing) = ...`, it's expecting to take whatever's on the rhs instead of borrowing it. `ref` there basically says you want to borrow the value temporarily. If the code compiles without the `clone()`, you're good to go!